plugin/src/test/java/io/jenkins/plugins/forensics/git/reference/GitParentCommitITest.java

```diff
@@ -0,0 +1,123 @@
+        Run<?, ?> lastBuild = buildSuccessfully(job);
+        GitCommitsRecord lastRecord = lastBuild.getAction(GitCommitsRecord.class);
+        ReferenceBuild lastRefBuild = lastBuild.getAction(ReferenceBuild.class);
+        assertThat(lastRecord).isNotNull()
+                .hasParentCommit(currentCommit);
+        assertThat(lastRefBuild).isNotNull()
+                .hasReferenceBuildId(build.getExternalizableId());
+    }
+
+    @Test
+    public void testMultiBranch() throws IOException {
+        // Init commit
+        String currentCommit = getHead();
+        FreeStyleProject job = createFreeStyleProject(JOB_NAME);
+
+        // root build, should have no reference build
+        Run<?, ?> masterBuild = buildSuccessfully(job);
+
+
+        checkoutNewBranch("side");
+        writeFileAsAuthorBar("Added new file to the side branch");
+
+        // parent commit should point to root build
+        Run<?, ?> sideBuild = buildSuccessfully(job);
+        GitCommitsRecord sideRecord = sideBuild.getAction(GitCommitsRecord.class);
+        ReferenceBuild sideRefBuild = sideBuild.getAction(ReferenceBuild.class);
+        assertThat(sideRecord).isNotNull()
+                .hasParentCommit(currentCommit);
+        assertThat(sideRefBuild).isNotNull()
+                .hasOwner(sideBuild)
+                .hasReferenceBuildId(masterBuild.getExternalizableId());
+
+
+        checkout("master");
+        writeFileAsAuthorBar("Edited file on master branch 1");
+
+        // parent commit should still point to root build, as the master branch now has an available build
+        Run<?, ?> masterBuild2 = buildSuccessfully(job);
+        GitCommitsRecord masterRecord2 = masterBuild2.getAction(GitCommitsRecord.class);
+        ReferenceBuild masterRefBuild2 = masterBuild2.getAction(ReferenceBuild.class);
+        assertThat(masterRecord2).isNotNull()
+                .hasParentCommit(currentCommit);
+        assertThat(masterRefBuild2).isNotNull()
+                .hasOwner(masterBuild2)
+                .hasReferenceBuildId(masterBuild.getExternalizableId());
+
+
+        // new branch feat has now the latest commit from the side branch as parent
+        checkout("side");
+        checkoutNewBranch("feat");
+        writeFileAsAuthorBar("side2 branch created and new file added");
+
+        // The reference build should now point to the second build, as the parent commit is from the branch side
+        Run<?, ?> featBuild = buildSuccessfully(job);
+        GitCommitsRecord featRecord = featBuild.getAction(GitCommitsRecord.class);
+        ReferenceBuild featRefBuild = featBuild.getAction(ReferenceBuild.class);
+        assertThat(featRecord).isNotNull()
+                .hasParentCommit(sideRecord.getLatestCommit());
+        assertThat(featRefBuild).isNotNull()
+                .hasOwner(featBuild)
+                .hasReferenceBuildId(sideBuild.getExternalizableId());
+    }
```

Comment:
It would be helpful to have another test with these steps:
```
- commit #1 master
- checkout new branch side 
- checkout master
- commit #2 master 
- build master (#1, contains commits #1 and #2)
- checkout side
- commit #1 side
- build side (#2) 
--> Then the parent build is not the latest commit of #1
```

Reply:
Ideally every commit should be built, as otherwise it would be a bit more difficult finding a matching reference build.

Reply:
I have updated the parent reference finder, however i am still discussing with my supervisor these changes